### ISSUE TITLE
AWS Data-Collector Batch stuck on running (Multiple instances created and not terminated)

### ISSUE THREAD

**Issue by luizfelipepg** [tmobile/pacbot#347]
I deployed pacbot to my AWS account and I am having a problem with the Data-Collector Batch process. Every day, multiple of them get started and got stuck showing this logs:

`[main] ERROR c.t.c.p.i.InventoryFetchOrchestrator - Delete Failed
com.amazonaws.services.s3.model.AmazonS3Exception: The XML you provided was not well-formed or did not validate against our published schema (Service: Amazon S3; Status Code: 400; Error Code: MalformedXML; Request ID: 6F0E1817BD592070; S3 Extended Request ID: vMzvm3LpBFUxMe1t4QdxZckvBdrIf2F2uKd15TuQcqQEH37w0fsTfrXYU8DUs2aMrESVZbiUaUA=)
`

Just after this I can see this lines I can see

```
2019-10-16 22:03:03 [main] INFO c.t.c.p.i.InventoryFetchOrchestrator - End : Backup Current Files
2019-10-16 22:03:03 [main] INFO c.t.c.p.i.InventoryFetchOrchestrator - Start : Upload Files to S3
2019-10-16 22:03:04 [main] INFO c.t.c.p.i.InventoryFetchOrchestrator - Uploading files to bucket: xxxxxx  folder: inventory
2019-10-16 22:03:07 [main] DEBUG c.t.c.p.i.InventoryFetchOrchestrator - Transfer % Completed :0.0
2019-10-16 22:03:10 [main] DEBUG c.t.c.p.i.InventoryFetchOrchestrator - Transfer % Completed :0.0
2019-10-16 22:03:13 [main] DEBUG c.t.c.p.i.InventoryFetchOrchestrator - Transfer % Completed :0.0
2019-10-16 22:03:16 [main] DEBUG c.t.c.p.i.InventoryFetchOrchestrator - Transfer % Completed :0.0
2019-10-16 22:03:19 [main] DEBUG c.t.c.p.i.InventoryFetchOrchestrator - Transfer % Completed :0.0
.
.
.
.
.
.
```

If you look at the Batch page, you can see everything is just stuck there:

<img width="1156" alt="66790493-a9a85100-ef3b-11e9-9f09-fe094203e1a9" src="https://user-images.githubusercontent.com/863641/67327026-05d62b00-f563-11e9-835d-7428d350f533.png">
<img width="1184" alt="66790496-ac0aab00-ef3b-11e9-9cfc-3ff2a9ae31ce" src="https://user-images.githubusercontent.com/863641/67327034-079fee80-f563-11e9-88e3-ba5d53ce6bf3.png">
*There are more than what it shows on those SS.

Some Data-Collector, by the way, run successfully. 
This problem results on a huge number of EC2 instances to start and never terminate. I got to a point where I had over 50 instances running. I have set this to only 1 account so far and I am afraid to add the other 10+ accounts I have now and this problem increase. 
This is also impacting on my billing as I have 10+ instances running for no reason overnight and I have to force terminate that job every morning.

Has anyone had any similar problem or can anyone help me with that ?

THanks :)

**Comment by kaykumar** [comment]
I think the inventory files are not getting uploaded to S3 for some reason, we will look into this, this is the first time we have seen this though, did you deploy from master or 1.6.1 release ?

**Comment by luizfelipepg** [comment]
I follow the steps on the page https://github.com/tmobile/pacbot/wiki/Install or at least I tried to. I could have missed or done something wrong. Here is the git log:

git log
commit 60083f9aeb345269cdacd5854d1eeb8d85e820a2 (HEAD -> master, tag: 1.6, origin/master, origin/HEAD)
Merge: 0ad0efe 0a37fbe
Author: Kamal Kumar <kumar.kamal@gmail.com>
Date:   Fri Sep 27 12:32:31 2019 -0700

    Merge pull request #335 from tmobile/release/1.6

    Release/1.6

**Comment by luizfelipepg** [comment]
@kaykumar , anything you believe I could do here to avoid this to happen ? I am running something to force stop batches every X hours but that is not ideal :/

**Comment by luizfelipepg** [comment]
@kaykumar @santhoshigorle @johnakash 
have you guys have any idea on what might be happening ?
I am happy to help with anything I can

**Comment by santhoshigorle** [comment]
we investigating issue and keep you updated as soon as we find root-cause.

**Comment by santhoshigorle** [comment]
The issue due to Aws SDK version conflict and to fix it you need to update the version in 2 files.  Please let us know if you need any additional help..

You can update the version and build the jars and upload to S3 bucket and the jobs would run fine in next invocation.

<img width="1057" alt="file1" src="https://user-images.githubusercontent.com/27745378/68886688-86083000-06cc-11ea-9119-0b1684f4c13e.png">

**Comment by MaossCO** [comment]
Hi,

Im having the same issue, but as i dont have that much experience with maven and java at all could you please explain in more detail how to rebuild the jars, i already updated the pom files with the correct version.

Thank you!

**Comment by sajeer-nooh** [comment]
Hi @MaossCO 
You can redeploy it by running the below command

**sudo python3 manager.py redeploy**

**Comment by anilcs81** [comment]
Fix is moved in 2.0
